src/LanguageServer/Protocol/Handler/LanguageServerNotificationManager.cs

```diff
@@ -3,38 +3,100 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Diagnostics.CodeAnalysis;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CommonLanguageServerProtocol.Framework;
 using StreamJsonRpc;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.Handler;
 
 internal class ClientLanguageServerManager : IClientLanguageServerManager
 {
     private readonly JsonRpc _jsonRpc;
+    private readonly AbstractLspLogger _logger;
 
-    public ClientLanguageServerManager(JsonRpc jsonRpc)
+    public ClientLanguageServerManager(JsonRpc jsonRpc, AbstractLspLogger logger)
     {
-        if (jsonRpc is null)
+        _jsonRpc = jsonRpc;
+        _logger = logger;
+    }
+
+    public async Task<TResponse?> SendRequestAsync<TParams, TResponse>(string methodName, TParams @params, CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+        try
+        {
+            return await _jsonRpc.InvokeWithParameterObjectAsync<TResponse>(methodName, @params, cancellationToken).ConfigureAwait(false);
+        }
+        catch (Exception ex) when (ConnectionLostFilter(ex, methodName))
         {
-            throw new ArgumentNullException(nameof(jsonRpc));
         }
 
-        _jsonRpc = jsonRpc;
+        return default;
```

Comment:
I'm not the biggest fan of this.  It would be nice there were a way to bubble up the exception, but ensure it doesn't get reported as a NFW.  Wonder if we should support adding a tag in the `Data` on the exception that the NFW reporter could look at and skip reporting if present?  cc @jasonmalinowski